### PR TITLE
feat: telemetry install and new endpoint for console

### PR DESCRIPTION
**Pull request by CristhianF7**
- Adds a new telemetry endpoint to send click events from Console
- Adds a Kubefirst heartbeat

Note: 3 new env vars have been added for telemetry `"CLUSTER_ID", "CLUSTER_TYPE", "INSTALL_METHOD"`

### CODE REVIEW

**Review by jarededwards — APPROVED** [review]

File: internal/router/api/v1/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/gin-gonic/gin"
+	"github.com/kubefirst/kubefirst-api/internal/db"
+	"github.com/kubefirst/kubefirst-api/internal/telemetryShim"
+	"github.com/kubefirst/kubefirst-api/internal/types"
+	log "github.com/sirupsen/logrus"
+)
+
+// PostTelemetry godoc
+// @Summary Create a Telemtry Event
```
Comment (echoboomer) [inline internal/router/api/v1/telemetry.go:21]: Spelling error on telemetry

File: internal/router/api/v1/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+// @Description Create a Telemtry Event
+// @Tags cluster
```
Comment (echoboomer) [inline internal/router/api/v1/telemetry.go]: Tag should be telemetry

File: internal/router/api/v1/telemetry.go (hunk 1 above)
Comment (echoboomer) [inline internal/router/api/v1/telemetry.go]: Spelling error on telemetry

File: internal/router/api/v1/telemetry.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+// @Accept json
+// @Param	cluster_name	path	string	true	"Cluster name"
+// @Param	definition	body	types.ClusterDefinition	true	"Cluster create request in JSON format"
+// @Success 202 {object} types.TelemetryRequest
```
Comment (echoboomer) [inline internal/router/api/v1/telemetry.go]: The telemetry request should actually go in the params section, the response should follow the typical success response

File: internal/router/api/v1/telemetry.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	clusterName, param := c.Params.Get("cluster_name")
+	if !param {
+		c.JSON(http.StatusBadRequest, types.JSONFailureResponse{
+			Message: ":cluster_name not provided",
+		})
+		return
+	}
+
+	// Create
```
Comment (echoboomer) [inline internal/router/api/v1/telemetry.go]: I don't think we need this check anymore for cluster existence

**Review by echoboomer — CHANGES_REQUESTED** [review]

**Review by echoboomer — APPROVED** [review]

**Pull request merged**